src/cogent3/core/new_alignment.py

```diff
@@ -901,16 +902,13 @@ def to_json(self):
         return json.dumps(self.to_rich_dict())
 
     def degap(self) -> SequenceCollection:
-        """Returns new collection in which sequences have no gaps.
+        """Returns new collection in which sequences have no gaps or missing
+        characters.
 
         Notes
         -----
         The returned collection will not retain an annotation_db if present.
```

Comment:
I don't think degapping should always remove a bound annotation_db. Add a design comment for an optional argument so a user can indicate whether to keep annotations.